### PR TITLE
[SPARK-41237][SQL] Reuse the error class `UNSUPPORTED_DATATYPE` for `_LEGACY_ERROR_TEMP_0030`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

The PR proposes to re-use the error class `UNSUPPORTED_DATATYPE ` for `_LEGACY_ERROR_TEMP_0030`.

### Why are the changes needed?

We should assign proper name to `_LEGACY_ERROR_TEMP_*` error classes.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

```
./build/sbt “sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*”
```

### CODE REVIEW

**Review by MaxGekk — COMMENTED** [review]
Can't you re-use the existing error class `UNSUPPORTED_DATATYPE`?

**Comment by itholic** [comment]
> Can't you re-use the existing error class `UNSUPPORTED_DATATYPE`?

Oh... I missed that one. Sounds good!

**Comment by itholic** [comment]
Btw, we use both `DATA_TYPE` and `DATATYPE` for error class name.
Maybe do we want to consolidate them in one rule ?

**Comment by MaxGekk** [comment]
> Maybe do we want to consolidate them in one rule ?

I agree. Let's use consolidate. I would prefer `DATATYPE` since it is shorter ;-)

**Comment by MaxGekk** [comment]
@itholic Could you fix test failures. Seems like they are related to your changes:
```
DataTypeParserSuite.Do not print empty parentheses for no params
org.scalatest.exceptions.TestFailedException: "
[UNSUPPORTED_DATATYPE] Unsupported data type unknown(line 1, pos 0)

== SQL ==
unknown
^^^
" did not contain "unknown is not supported"
```

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala [hunk 1]
```diff
@@ -292,8 +292,8 @@ private[sql] object QueryParsingErrors extends QueryErrorsBase {
 
   def dataTypeUnsupportedError(dataType: String, ctx: PrimitiveDataTypeContext): Throwable = {
     new ParseException(
-      errorClass = "_LEGACY_ERROR_TEMP_0030",
-      messageParameters = Map("dataType" -> dataType),
+      errorClass = "UNSUPPORTED_DATATYPE",
+      messageParameters = Map("typeName" -> dataType),
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryParsingErrors.scala:296]: Need to quote the type even if it is incorrect. Please, call `toSQLType`.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Pull request closed without merging**